lib/transloadit/TransloaditRequest.php

```diff
@@ -78,6 +80,38 @@ public function execute($response = null) {
     $this->prepare();
     $response = parent::execute(new TransloaditResponse());
     $response->parseJson();
+
+    if ($this->path === '/assemblies' && $this->waitForCompletion) {
+      return $this->_waitForCompletion($response);
+    }
     return $response;
   }
+
+  private function _waitForCompletion($response) {
+    $assemblyUrl = $response->data['assembly_ssl_url'];
+    $parts = parse_url($assemblyUrl);
+
+    while (true) {
+      $req = new TransloaditRequest();
+      $req->endpoint = 'https://' . $parts['host'];
+      $req->path = $parts['path'];
+      $response = $req->execute();
+
+      if ($response->data['ok'] === 'ASSEMBLY_COMPLETED') {
+        return $response;
+      }
+
+      if (isset($response->data['error']) && !empty($response->data['error'])) {
```

Comment:
I think that the if statements could be simplified a bit. We only need to wait if the status is uploading or executing. In all other cases, we can return the response. There is no distinction between a successful finish and an assembly failure, is there?

Otherwise, the PR looks good